adbms/src/test/scala/de/up/hpi/informationsystems/adbms/definition/DactorTest.scala

```diff
@@ -31,22 +31,19 @@ object DactorTest {
   object DactorWithRelation {
     def props(id: Int): Props = Props(new DactorWithRelation(id))
 
-    object TestRelationDef {
-      val name: String = "testrelation"
+    object TestRelationDef extends RelationDef {
       val col1: ColumnDef[Int] = ColumnDef("col1")
       val col2: ColumnDef[String] = ColumnDef("col2")
-      val columns: Set[UntypedColumnDef] = Set(col1, col2)
 
-      def rowRelation: MutableRelation = new RowRelation {
-        override val columns: Set[UntypedColumnDef] = TestRelationDef.columns
-      }
+      override val columns: Set[UntypedColumnDef] = Set(col1, col2)
+      override val name: String = "testrelation"
```

Comment:
Why is the name here `testrelation` while it is known to its `Dactor` as `testRelation`?
And does a `RelationDef` really need a name?

Reply:
it's just a test and I haven't thought about it very much :laughing: 
I will change this.

I already thought of introducing a declared name, which you can later use to send the insert messages to. Like I did in `DactorTest`:

```scala
"insert matching record successfully" in {
      val insertMessage = DefaultMessagingProtocol.InsertIntoRelation(
        TestRelationDef.name,     // <-- here
        Seq(
          Record(TestRelationDef.columns)(
            TestRelationDef.col1 ~> 1 &
            TestRelationDef.col2 ~> "1"
          ).build()
      ))
      dut.tell(insertMessage, probe.ref)
      probe.expectMsg(akka.actor.Status.Success)
}
```

I can imagine to let the `Dactor` handle the naming of its relations, as it is currently the only one using it:

```scala
class MyDactor(id: Int) extends Dactor(id) {
  val firstRel = RowRelation(RelDef1)
  val secondRel = RowRelation(RelDef2)

  override val declareRelations: Set[MutableRelation] = Set(firstRel, secondRel)

  // in `Dactor` trait:
  val relations: Map[String, MutableRelation] = declareRelations.map{ rel =>
    rel.name -> rel
  }.toMap
}
```
Then you are sure, that you can send messages to a `Dactor` and specify the relation by its name `DactorComp.RelDef.name`.